Add warning that actions require ament_cmake, not ament_python #4986 (#5201)
ros2/ros2_documentation: source/Tutorials/Intermediate/Creating-an-Action.rst

1 Creating an interface package
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. warning::

>    Actions can only be created in a C++ package using ``ament_cmake``. They are not supported in Python packages using ``ament_python``. This restriction also applies to messages and services, so ensure your package is configured as a C++ package before proceeding with this tutorial.
> 

Reviewer: ```suggestion
   Interfaces for Actions (along with Message and Service) can only be generated by a package which has ``ament_cmake`` as its ``build_type``.

```

.. note::

>    It is recommended to define action interfaces (as well as messages and services) in a separate package dedicated to interfaces, as shown in this tutorial. This improves modularity and makes it easier to reuse them across multiple projects.

Reviewer 2: one complete sentence per line. i think this line is detected warnig by our line checker, once it is approved(start the github action).

```suggestion
   It is recommended to define action interfaces (as well as messages and services) in a separate package dedicated to interfaces, as shown in this tutorial.
   This improves modularity and makes it easier to reuse them across multiple projects.
```

> 

Reviewer: I suggest to move these two statements above the warning and outside the `note` block (ie, standalone statements) right after the `Creating an interface package` header.

Author: okay

1 Creating an interface package
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Actions can only be created in a package using `ament_cmake`. They are not supported

> in Python packages using `ament_python`. This restriction also applies to messages and services, so

Reviewer 3: ```suggestion
in Python packages using `ament_python`. 
This restriction also applies to messages and services, so
```

1 Creating an interface package
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Actions can only be created in a package using `ament_cmake`. They are not supported

Reviewer 3: ```suggestion
Actions can only be created in a package using `ament_cmake`. 
They are not supported in Python packages using `ament_python`.
```

> Actions can only be created in a C++ package using ``ament_cmake``. They are not supported

Reviewer 2: I do not think this is true. it does not have to be C++ package, rather it has to be CMake package.

> Actions can only be created in a C++ package using ``ament_cmake``. They are not supported

Reviewer 3: ```suggestion
Actions can only be created in a C++ package using ``ament_cmake``.
They are not supported in Python packages using ``ament_python``.
```

> Actions can only be created in a package using `ament_cmake`. 
> They are not supported in Python packages using `ament_python`.

Reviewer 4: You need two backticks for `literal formatting` (compared to just 1 in Markdown), see https://docs.ros.org/en/rolling/The-ROS2-Project/Contributing/Contributing-To-ROS-2-Documentation.html#code-formatting

```suggestion
Actions can only be created in a package using ``ament_cmake``. 
They are not supported in Python packages using ``ament_python``.
```

> This restriction also applies to messages and services, so

Reviewer 2: this sentence looks like cut in the middle?

```suggestion
This restriction also applies to messages and services, so...
```

> in Python packages using ``ament_python``. This restriction also applies to messages and services, so

Reviewer 3: ```suggestion
This restriction also applies to messages and services, so ensure your package is configured as a C++ package before proceeding with this tutorial.
```

.. note::

>    It is recommended to define action interfaces in this tutorial.

Reviewer 4: Sorry for adding yet another comment on this PR, but I'm not sure I understand what this means. Do you mean this?

```suggestion
   It is recommended to define action interfaces in their own package.
```

> ensure your package is configured as a C++ package before proceeding with this tutorial.

Reviewer 3: ```suggestion
```

>    It is recommended to define action interfaces in this tutorial. This improves modularity and makes it

Reviewer 3: ```suggestion
   It is recommended to define action interfaces in this tutorial. 
   This improves modularity and makes it easier to reuse them across multiple projects.
```

>    easier to reuse them across multiple projects.

Reviewer 3: ```suggestion
```

1 Creating an interface package
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Actions can only be created in a package using `ament_cmake`. 
They are not supported in Python packages using `ament_python`.

> in Python packages using `ament_python`. 

Reviewer 3: ```suggestion
```

Next steps
----------

> Next, let's utilize your newly defined action interface by creating an action server
> and client (in :doc:`Python <Writing-an-Action-Server-Client/Py>` or :doc:`C++ <Writing-an-Action-Server-Client/Cpp>`).

Reviewer 2: ```suggestion
Next, let's utilize your newly defined action interface by creating an action server and client (in :doc:`Python <Writing-an-Action-Server-Client/Py>` or :doc:`C++ <Writing-an-Action-Server-Client/Cpp>`).
```

Related content
---------------

For more detailed information about ROS actions, please refer to the `design article <https://design.ros2.org/articles/actions.html>`__.

> [blank line]

Reviewer 2: what is this?

```suggestion
[blank line]
```

Related content
---------------

> For more detailed information about ROS actions, please refer to the `design article <http://design.ros2.org/articles/actions.html>`__.

Reviewer 3: ```suggestion
For more detailed information about ROS actions, please refer to the `design article <http://design.ros2.org/articles/actions.html>`__.

```
